src/fb.coffee

```diff
@@ -115,7 +122,28 @@ class FBMessenger extends Adapter
             @robot.emit "fb_richMsg", envelope
             @_processAttachment event, envelope, attachment for attachment in envelope.attachments
         if event.message.text?
-            @receive new TextMessage envelope.user, event.message.text
+            text = @cleanMessageText event.message.text, envelope.room
+            msg = new TextMessage envelope.user, text, event.message.mid
+            @receive msg
+            @robot.logger.info "Reply message to room/message: " + envelope.user.name + "/" + event.message.mid
+
+    cleanMessageText: (text, chat_id) ->
+        # if AUTOHEAR is not turned on, don't clean
+        if !process.env['FB_AUTOHEAR']
+            return text
+        
+        # If it is a private chat, automatically prepend the bot name if it does not exist already.
+        if (chat_id > 0)
+            # Strip out the stuff we don't need.
+            text = text.replace(/^\//g, '').trim()
+
+            text = text.replace(new RegExp('^@?' + @robot.name.toLowerCase(), 'gi'), '');
+            text = text.replace(new RegExp('^@?' + @robot.alias.toLowerCase(), 'gi'), '') if @robot.alias
+            text = @robot.name + ' ' + text.trim()
+        else
+            text = text.trim()
```

Comment:
iirc messenger already trims for you.  in any case, not sure this belongs in adapter code -- if someone wants messages to be trimmed, they can do that using [response middleware](https://github.com/github/hubot/pull/1021).  we want to avoid manipulating content in messages as little as possible since that can really be the source of mysterious bugs/unintended behavior